File: Java/GermanAddressBuilder.java
public class GermanAddressBuilder extends AddressBuilder {

    public GermanAddressBuilder(AddressData addressData) {
        super(addressData);
        titleF = "Frau";
        titleM = "Herrn";
    }

    @Override
    public void buildLine2(){
        address.addLine(addressData.street() + " " + addressData.number() + "\n");
    }

    @Override
    public void buildLine3(){
        address.addLine(addressData.zip() + " " + addressData.city() + "\n");
    }
}

File: TypeScript/GermanAddressBuilder.ts
import { AddressBuilder } from './AddressBuilder';
import { AddressData } from './AddressData';

export class GermanAddressBuilder extends AddressBuilder{

  buildLine2() : void {
    this.address.addLine(this.addressData.street + " " + this.addressData.number + "\n");
  }

  buildLine3() : void {
    this.address.addLine(this.addressData.zip + " " + this.addressData.city + "\n");
  }

  constructor(addressData: AddressData) {
    super(addressData);
    this.title_f = "Frau";
    this.title_m = "Herrn";
  }
}